hosts/gitlab.go

```diff
@@ -22,12 +22,23 @@ type Gitlab struct {
 
 // NewGitlabClient accepts a manager struct and returns a Gitlab host pointer which will be used to
 // perform a gitlab audit on an group or user.
-func NewGitlabClient(m manager.Manager) *Gitlab {
-	return &Gitlab{
+func NewGitlabClient(m manager.Manager) (*Gitlab, error) {
+	var err error
+
+	gitlabClient := &Gitlab{
 		manager: m,
 		ctx:     context.Background(),
 		client:  gitlab.NewClient(nil, options.GetAccessToken(m.Opts)),
 	}
+
+	if m.Opts.BaseURL != "" {
+		err := gitlabClient.client.SetBaseURL(m.Opts.BaseURL)
+		if err != nil {
+			log.Error(err)
```

Comment:
This check could be removed since the error will be bubbled up in main.go

Reply:
Updated